src/execution/execution.cc

```diff
@@ -352,27 +376,27 @@ V8_WARN_UNUSED_RESULT MaybeHandle<Object> Invoke(Isolate* isolate,
       // User JS should not get executed in divergent code paths,
       // unless we have paused.
       // → Print log and prevent execution.
-      Script::PositionInfo info;
-      std::string name;
-      Handle<Script> script;
-      if (function->shared().script().IsScript()) {
-        script = Handle<Script>(Script::cast(function->shared().script()), isolate);
-        Script::GetPositionInfo(script, function->shared().StartPosition(),
-                                &info, Script::WITH_OFFSET);
-        name = script->name().IsString()
-                   ? String::cast(script->name()).ToCString().get()
-                   : "(anonymous script)";
-      }
+      std::string location = GetFunctionLocationInfo(isolate, function);
       std::stringstream stack;
       isolate->PrintCurrentStackTrace(stack);
 
       recordreplay::Warning(
-          "JS Invoke: Non-deterministic user JS PC=%zu scriptId=%d @%s:%d:%d stack=%s",
-          *gProgressCounter, script.is_null() ? script->id() : -1, name.c_str(), info.line + 1,
-          info.column, stack.str().c_str());
+          "JS Invoke: Non-deterministic user JS PC=%zu %s stack=%s",
+          *gProgressCounter, location.c_str(), stack.str().c_str());
       return isolate->factory()->undefined_value();
     }
 
+    if (recordreplay::IsReplaying() &&
+        IsMainThread() &&
+        !recordreplay::AreEventsDisallowed() &&
+        RecordReplayDependencyGraphExecutionDepth() == 0) {
+      static bool emit = getenv("RECORD_REPLAY_WARN_MISSING_DEPENDENCY_GRAPH");
+      if (emit) {
+        std::string location = GetFunctionLocationInfo(isolate, function);
+        recordreplay::Warning("Missing dependency graph execution: %s", location.c_str());
```

Comment:
Can you give it a label in front? This way, its easier to find all "dependency graph" warnings using our `one_word` or `two_word` HC filter (e.g. [this](https://ui.honeycomb.io/replay/datasets/backend/usage/result/a4nzbkakEWt)).